src/EntityFramework/Migrations/Infrastructure/EdmModelDiffer.cs

```diff
@@ -328,6 +328,27 @@ where m1.EdmEquals(m2)
                     / (entityType1.DeclaredMembers.Count + entityType2.DeclaredMembers.Count)) > 0.80;
         }
 
+        private bool SourceAndTargetMatch(EntityType sourceEntityType, EntityTypeMapping sourceEntityTypeMapping, EntityType targetEntityType, EntityTypeMapping targetEntityTypeMapping)
+        {
+            return (sourceEntityTypeMapping.EntityType != null
+                    && targetEntityTypeMapping.EntityType != null
+                    && sourceEntityType == sourceEntityTypeMapping.EntityType
+                    && targetEntityType == targetEntityTypeMapping.EntityType)
+                   || (sourceEntityTypeMapping.EntityType == null
+                       && targetEntityTypeMapping.EntityType == null
+                       && sourceEntityTypeMapping.IsOfTypes.Contains(sourceEntityType)
```

Comment:
How costly is `IsOfTypes`? Would it be better to unroll this horrifically complex boolean so we can read it and assign the `IsOfTypes` collection to a local variable?